core.d.ts

```diff
@@ -435,3 +438,88 @@ console.log(await fileTypeFromBlob(blob));
 ```
 */
 export declare function fileTypeFromBlob(blob: Blob): Promise<FileTypeResult | undefined>;
+
+/**
+Function that allows specifying custom detection mechanisms.
+
+An iterable of detectors can be provided via the fileTypeOptions argument for the {@link FileTypeParser.constructor}.
+
+The detectors are called before the default detections in the provided order.
+
+Custom detectors can be used to add new FileTypeResults or to modify return behaviour of existing FileTypeResult detections.
+
+If the detector returns `undefined`, there are 2 possible scenarios:
+
+	1. The detector has not read from the tokenizer, it will be proceeded with the next available detector.
+	2. The detector has read from the tokenizer (`tokenizer.position` has been increased).
+		 In that case no further detectors will be executed and the final conclusion is that file-type returns undefined.
+		 Note that this an exceptional scenario, as the detector takes the opportunity from any other detector to determine the file type.
```

Comment:
```suggestion
1. The detector has not read from the tokenizer, it will be proceeded with the next available detector.
2. The detector has read from the tokenizer (`tokenizer.position` has been increased).
	In that case no further detectors will be executed and the final conclusion is that file-type returns undefined.
	Note that this an exceptional scenario, as the detector takes the opportunity from any other detector to determine the file type.
```